index.js

```diff
@@ -44,48 +45,85 @@ function cleanClock (clock) {
 }
 
 exports.init = function (sbot, config) {
-  const dir = config.path ? path.join(config.path, 'ebt') : null
-  const store = Store(dir, null, toUrlFriendly)
-
-  const ebt = EBT({
-    logging: config.ebt && config.ebt.logging,
-    id: sbot.id,
-    getClock (id, cb) {
-      store.ensure(id, function () {
-        const clock = store.get(id) || {}
-        cleanClock(clock)
-        cb(null, clock)
-      })
-    },
-    setClock (id, clock) {
-      cleanClock(clock, 'non-feed key when saving clock')
-      store.set(id, clock)
-    },
-    getAt (pair, cb) {
-      sbot.getAtSequence([pair.id, pair.sequence], (err, data) => {
-        cb(err, data ? data.value : null)
-      })
-    },
-    append (msg, cb) {
-      sbot.add(msg, (err, msg) => {
-        cb(err && err.fatal ? err : null, msg)
-      })
-    },
-    isFeed: isFeed
-  })
+  const formats = {
+    'classic': require('./formats/classic')
+  }
+
+  const ebts = {}
+  function addEBT(formatName) {
+    const dirName = 'ebt' + (formatName === 'classic' ? '' : formatName)
+    const dir = config.path ? path.join(config.path, dirName) : null
+    const store = Store(dir, null, toUrlFriendly)
+
+    const format = formats[formatName]
+    // EBT expects a function of only feedId so we bind sbot here
+    format.isFeed = (feedId) => format.sbotIsFeed(sbot, feedId)
```

Comment:
To solve the naming issue, you could use `fn.bind(thisArg, arg1, arg2, ...)` like this:

```js
function foo(a,b,c) { console.log(a,b,c) }

//                       this, a
const coolFoo = foo.bind(null, 10)

coolFoo('x','y') // 10 x y
```

So
```suggestion
    format.isFeed = format.isFeed.bind(format, sbot)
```

Reply:
Right, just what I was looking for. Thanks!